### ISSUE TITLE
Detect notebook language from metadata and use it to recommend extensions

### ISSUE THREAD

**Issue by SiddhanthUnnithan** [microsoft/vscode#132061]
## TL;DR
Use notebook metadata to recommend Jupyter **and** the right language extensions when the user clicks on the "Install kernels from marketplace" option.

## Overview
Today, the built-in support for notebooks surfaces an option to install recommended kernels from the marketplace when the user does not have extensions installed which contribute kernel options to run with.

When clicking on the option the user is shown the marketplace pane filtered to show the Jupyter extension. This is valuable as having the Jupyter extension installed is a pre-requisite for rendering some output (for now) and running code in the notebook. However, the user must also install the right language extension based on the kernel that they're trying to use. Today, a user that wishes to run Python code in a notebook can view Python kernels as listed by the Jupyter extension, but require the Python extension installed before running their code:

![image](https://user-images.githubusercontent.com/6413608/131701440-dc316040-0bd1-4b9e-a4eb-502bf2f7014e.png)

The built-in notebook support should use the notebook metadata to detect the language of the notebook and recommend a more "complete" set of extensions when the user chooses the "Install kernels from marketplace" option. Using Python as an example, if the built-in support detects the notebook language as Python, both the Jupyter and Python extension should be shown to the user in the marketplace filter.

**Comment by SiddhanthUnnithan** [comment]
- What % of notebooks are Python notebooks?
- Short term 1: Recommend Python extension instead of Jupyter
- Short term 2: Recommend Python and .NET
- Short term 3: Detect whether there is one Python cell in the notebook. If so, recommend Python extension. If not, only recommend Jupyter.

**Comment by greazer** [comment]
90%+ of github notebooks are Python (see @tanhakabir and @claudiaregio) 

We're already *temporarily* hardcoding the Jupyter extension as the recommended extension when performing any execution based operation on a notebook. Since running a Python notebook requires the Python extension to be installed (this is actually a long-term expectation as well) its seems more appropriate to recommend that extension instead of Jupyter in this scenario. In Sid's table above, it's option #1.

Option #2 needs some investigation to ensure it's viable. It's also a little more hacky since we'd be hardcoding more than one extension in the short term. However, it would be more satisfactory short-term solution for the non-Python notebook community. 

Option #3 could really be considered a short term solution for addressing this very issue. Since most notebooks are Python, making sure we detect at least one Python cell before recommending Python will have high bang for buck.
